ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -9562,6 +9568,59 @@ void HqlCppTranslator::doBuildExprFailCode(BuildCtx & ctx, IHqlExpression * expr
     }
 }
 
+void HqlCppTranslator::doBuildExprMatchedInJoin(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    HqlExprAssociation * matchedMarker = ctx.queryMatchExpr(activeJoinTransformFlagsMarkerExpr);
+    if (!matchedMarker)
+    {
+        if (!buildExprInCorrectContext(ctx, expr, tgt, false))
+            throwError(HQLERR_MatchedContextNotFound);
+        return;
+    }
+    __int64 mask = 0;
+    IHqlExpression * patternExpr = expr->queryChild(0);
+    if (patternExpr)
+    {
+        //shouldn't have been called if not left or right
+        IAtom *patternName = patternExpr->queryName();
+        if (patternName==leftAtom)
+            mask = JTFmatchedleft;
+        else if (patternName==rightAtom)
+            mask = JTFmatchedright;
+        else
+            throwError(HQLERR_InvalidMatchedPatternInJoin);
+    }
+    IValue * maskValue = createIntValue(mask, 4, false);
+    tgt.expr.setown(createValue(no_band, LINK(matchedMarker->queryExpr()), createConstant(maskValue)));
```

Comment:
Should include the type of the expression in createValue(), and should probably also compare against 0 to create an expression of boolean type.